connector/oidc/oidc.go

```diff
@@ -184,6 +187,11 @@ func (c *Config) Open(id string, logger log.Logger) (conn connector.Connector, e
 		c.PromptType = "consent"
 	}
 
+	pkceVerifier := ""
+	if c.EnablePKCE {
```

Comment:
Do you know if we need an option for this? We can rely on what the provider exposes via the discovery endpoint.

```
  "code_challenge_methods_supported": [
    "S256",
    "plain"
  ],
```

If the `code_challenge_methods_supported` section is present, we can use PKCE.

Reply:
@vixus0 my bad, I overlooked the info from your first post
> Dynamic configuration using the OAuth2 discovery metadata isn't possible as the coreos/go-oidc library currently doesn't expose code_challenge_methods_supported in the Provider.

Reply:
So, after checking the docs once again, there is an option to extract the provider's metadata by using https://pkg.go.dev/github.com/coreos/go-oidc/v3/oidc#Provider.Claims